smac/utils/validate.py

```diff
@@ -81,34 +82,50 @@ def __init__(self, scenario: Scenario, trajectory: list,
             num_run = np.random.randint(MAXINT)
             self.rng = np.random.RandomState(seed=num_run)
 
-    def _save_results(self, rh:RunHistory, output):
-        """ Helper to save results to file """
-        if output == "":
+    def _save_results(self, rh: RunHistory, output_fn, backup_fn=None):
+        """ Helper to save results to file
+
+        Parameters
+        ----------
+        rh: RunHistory
+            runhistory to save
+        output_fn: str
+            filename to save history to
+        backup_fn: str
+            if output_fn is not a file, treating as dir and append this filename
```

Comment:
Sorry for extreme nitpicking, but this is still not clear. Also, it contradicts the code below because it does not need to be a file to not create a new directory, but a string ending on `.json`.

Reply:
Don't be sorry, I should be more precise from the start. How does it look now?